### ISSUE TITLE
implement basic kernel printing functionality

### ISSUE THREAD

**Issue by ta5een** [ta5een/wyoos#26]
Depends on #25

Implement basic functionality of the following functions to be used in the kernel for logging and debugging purposes:

- `printf`, `snprintf`, `sprintf`, `vprintf`
- `putc`, `putchar`
- `puts`

Stream-oriented versions (such as `fprintf`) are out of scope until file I/O capabilities are implemented. As a result, the concepts of `stdout` and `stderr` will not be applicable and will not be implemented for now.

For now, the functions listed above should output all strings to the serial port and the VGA text buffer.

Future considerations:

- Implementing logging levels (debug, info, error, etc.) for filtering log messages. See #22.
- Ensuring thread safety in concurrent environments.
- Introducing buffering to optimise output performance.
